### PR TITLE
[Sofa.Type] Fix some compilation warnings

### PR DESCRIPTION
**Pull request by alxbilger**
Vector negation does not make sense for unsigned types

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/Sofa.Type/src/sofa/type/Vec.h [hunk 1]
```diff
@@ -422,12 +422,20 @@ class Vec : public sofa::type::fixed_array<ValueType,size_t(N)>
...
-    constexpr Vec<N,ValueType> operator-() const noexcept
+    constexpr Vec<N, ValueType> operator-() const noexcept
     {
-        Vec<N,ValueType> r(NOINIT);
-        for (Size i=0; i<N; i++)
-            r[i]=-this->elems[i];
-        return r;
+        if constexpr (std::is_unsigned_v<ValueType>)
+        {
+            static_assert("Trying to negate an unsigned vector");
```
Comment (fredroy) [inline SofaKernel/modules/Sofa.Type/src/sofa/type/Vec.h]: Should we stop the compilation if we try to negate a vector of unsigned ?

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/Sofa.Type/src/sofa/type/Vec.h (hunk 1 above)
Comment (alxbilger) [reply SofaKernel/modules/Sofa.Type/src/sofa/type/Vec.h]: Yes, that's the idea

**Comment by fredroy** [comment]
I dont know if it is linked with this PR but one additional scenes is failing `SceneTests.examples/Benchmark/TopologicalChanges.ProjectDirectionConstraint_RemovingMeshTest.scn` ?

**Comment by alxbilger** [comment]
@fredroy This scene crashes randomly but rarely on master. So I don't think it's related to this PR.

**Pull request merged**